presto-main/src/main/java/com/facebook/presto/server/TaskResource.java

```diff
@@ -170,18 +170,14 @@ public Response deleteTask(@PathParam("taskId") TaskId taskId,
     {
         requireNonNull(taskId, "taskId is null");
 
-        TaskInfo taskInfo;
         if (abort) {
-            taskInfo = taskManager.abortTask(taskId);
+            taskManager.abortTask(taskId);
         }
         else {
-            taskInfo = taskManager.cancelTask(taskId);
+            taskManager.cancelTask(taskId);
         }
 
-        if (shouldSummarize(uriInfo)) {
-            taskInfo = taskInfo.summarize();
-        }
-        return Response.ok(taskInfo).build();
+        return Response.ok().build();
```

Comment:
IIRC you can just change the return type to `void` and return nothing.  Then JAX-RS will return a 204 no content